src/QMCWaveFunctions/IonOrbitalBuilder.cpp

```diff
@@ -28,6 +28,8 @@ bool
 IonOrbitalBuilder:: put(xmlNodePtr cur)
 {
   ParticleSet &p = targetPtcl;
+  widthOpt.resize(targetPtcl.getTotalNum());
+  widthOpt = -1;
```

Comment:
Why initialize the value to -1? The later non-zero checking seems assume initialized value 0 and requires full user input.

Reply:
@ye-luo the width parameter is required. The intended behavior is abort when no user input is given.

Reply:
I agree that the intended behaviour is abort when no user input is given. But I'm not convinced that it justifies your change. The old code relies on oAttrib to do the resize of widthOpt if a user input is give. If no input is given. widthOpt has zero size and abort. In your change, you resize and initialize the value to -1. The following num_nonzero will not be able to abort if no user input is given. If you set the initial value to 0, I may agree.

Reply:
@ye-luo "The old code relies on oAttrib to do the resize of widthOpt" is exactly why IonOrbital has been broken. For some reason `oAttrib` no longer automatically does the resize. I see your point about the change in initial default now. I will change the defaults to 0.